diff --git a/search_api_page.pages.inc b/search_api_page.pages.inc
index 654a7cd..e934d0a 100644
--- a/search_api_page.pages.inc
+++ b/search_api_page.pages.inc
@@ -22,7 +22,7 @@ function search_api_page_view($id, $keys = NULL, $overridden_options = array())
   }
 
   // Merge in overridden options.
-  if ($overridden_options) {
+  if (is_array($overridden_options)) {
     $page->original_options = $page->options;
     $page->options = $overridden_options + $page->options;
   }
