From f8c90eadf09c743ef1e65c58001df70db2dd7855 Mon Sep 17 00:00:00 2001
From: thePanz <thepanz@gmail.com>
Date: Wed, 26 Feb 2014 15:29:39 +0100
Subject: [PATCH] Allow non-pdf files to be uploaded (bypass image preview
 creation)

---
 pdf_to_image.module | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git pdf_to_image.module pdf_to_image.module
index 980de60..0c01978 100644
--- pdf_to_image.module
+++ pdf_to_image.module
@@ -314,7 +314,7 @@ function pdf_to_image_field_widget_form(&$form, &$form_state, $field, $instance,
 
   foreach (element_children($elements) as $delta) {
     // If not using custom extension validation, ensure this is a pdf.
-    $elements[$delta]['#upload_validators']['file_validate_extensions'][0] = 'pdf';
+    // $elements[$delta]['#upload_validators']['file_validate_extensions'][0] = 'pdf';
     // File fields need extra processing. Our one even more so.
     #$elements[$delta]['#process'][] = 'pdf_to_image_field_widget_process';
   }
@@ -392,6 +392,11 @@ function pdf_to_image_generate_process($entity_type, $entity, $field_id, $field_
     return;
   }
   $pdf_file = file_load($items[$field_lang][0]['fid']);
+  // Skip non-PDF files
+  if ($pdf_file->filemime != 'application/pdf') {
+    return;
+  }
+
   $pdf_realpath = file_stream_wrapper_get_instance_by_uri($pdf_file->uri)->realpath();
   $count = pdf_to_image_count_pages($pdf_realpath);
   if (!empty($field_instance['widget']['settings']['pdf_to_image']['target_field'])) {
-- 
1.8.4.msysgit.0

