diff --git a/template.php b/template.php
index 559ab3d..ff93b6d 100644
--- a/template.php
+++ b/template.php
@@ -318,6 +318,9 @@ function zen_preprocess_node(&$variables, $hook) {
   // Add $unpublished variable.
   $variables['unpublished'] = (!$variables['status']) ? TRUE : FALSE;
 
+  // Set preview variable to FALSE if it doesn't exist.
+  $variables['preview'] = isset($variables['preview']) ? $variables['preview'] : FALSE;
+
   // Add pubdate to submitted variable.
   $variables['pubdate'] = '<time pubdate datetime="' . format_date($variables['node']->created, 'custom', 'c') . '">' . $variables['date'] . '</time>';
   if ($variables['display_submitted']) {
